packages/style-engine/class-wp-style-engine.php

```diff
@@ -230,6 +230,26 @@ public static function get_instance() {
 		return self::$instance;
 	}
 
+	/**
+	 * Get an instance of the WP_Style_Engine_CSS_Rules object.
+	 *
+	 * If a $selector is provided, then the same instance will be used each time the same $selector is used,
+	 * otherwise, a new instance will be created each time.
+	 *
+	 * @param string $selector The selector to use for the CSS rules.
+	 * @return WP_Style_Engine_CSS_Rules The CSS rules object.
+	 */
+	public function get_css_rules_object( $selector = null ) {
+		if ( null === $selector ) {
+			return new WP_Style_Engine_CSS_Rules();
+		}
+		static $instances = array();
+		if ( ! isset( $instances[ $selector ] ) ) {
+			$instances[ $selector ] = new WP_Style_Engine_CSS_Rules();
```

Comment:
Oh, a separate store would also be testable 😇

Reply:
> In something like a collection class that exposes add, get, and delete methods.

Do you mean like a registry? There's already a familiar naming convention in the block type and pattern registries so there could potentially be a parallel there... I'm not sure how suitable that API design is to this use case, but just thought I'd mention it in case it helps predictability to have a common "pattern" there 🙂

Reply:
The `get_css_rules_object` method was something I added in lieu of a store.
I figured that having a method like that would be easy to extend, and in a future PR add a store which can be used without changing the interface a lot. So if we add a `store`, we can simply replace the contents of this function.
I'm trying to keep PRs like this one as small as possible so they're easier to review and convey the basic concepts/ideas 😉 
So I think of this as a primitive store/multiton implementation that will only be used for a short while until we add a store etc.

> Also, there are some styles that we might not want to "store", e.g., it's going to be a while before block supports move from inline style attributes. Therefore there'd have to be a way to bypass it.

That's one of the reasons I did it like this... 😆 If a `$selector` is not defined, then a new "unnamed" object is created and not stored in the array. It's effectively bypassing the store.

Reply:
> Do you mean like a registry?

I don't know 🤣 I suppose so. At the least some sort of interface to interact with the store. 

Given that we'll probably need to sort/fetch store items by layers and [insert latest CSS tech name here], we'll probably need something extendable down the track.

> I figured that having a method like that would be easy to extend, and in a future PR add a store which can be used without changing the interface a lot. So if we add a store, we can simply replace the contents of this function.

Ah, that makes sense, thanks. 

I suppose if it's testable then it's alright to have in the main class. It's also something we can play with later since there's no immediate need to store rules right now while we're still pulling things together.

Reply:
While I think of it, in terms of naming, I wonder if the class that contains the behaviour for storing / outputting by layers and/or in `style` tags etc, could be a `WP_Style_Engine_Renderer` to consolidate the various different pieces that are all to do with the final output?

Reply:
A "Renderer" would make sense to have in the final version of the Style Engine, yes 👍

Reply:
Yeah some sort of renderer sounds good. Maybe it's the glue method (retrieve, process, compile, enqueue) that we register with the WP hook (?)

Reply:
For inline styles (by _inline_ here I mean `<div style="...">` I don't think a renderer would play any significant role. But when we get to inline styles using a `<style>` element, then yes, I 100% agree that the renderer should be the one that gets registered in the WP hook. It can then gather all styles from the store that have a `selector` (and can therefore be printed in a `<style>` element, apply any optimizations that we may need, and then print them all together 👍

Reply:
The primitive store was removed 👍